From 81982d6e231915e80aac9121caf15d3a452b70da Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:33:06 -0400
Subject: [alsa] hda/realtek: Add support headset mode for ALC233

Message-id: <140972823573-307-patchset-jkysela@redhat.com>
Patchwork-id: 91003
O-Subject: [RHEL 7.1 PATCH 306/442] ALSA: hda/realtek - Add support headset mode for ALC233
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Signed-off-by: Kailang Yang <kailang@realtek.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 13fd08a339f174840046d0b229f434c0a5ee9925)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 1f06007..c0b16de 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -3524,6 +3524,7 @@ static void alc_headset_mode_unplugged(struct hda_codec *codec)
   /* Direct Drive HP Amp control */
   alc_write_coefex_idx(codec, 0x57, 0x03, 0x8aa6);
   break;
+ case 0x10ec0233:
  case 0x10ec0283:
   alc_write_coef_idx(codec, 0x1b, 0x0c0b);
   alc_write_coef_idx(codec, 0x45, 0xc429);
@@ -3582,6 +3583,7 @@ static void alc_headset_mode_mic_in(struct hda_codec *codec, hda_nid_t hp_pin,
   alc_write_coef_idx(codec, 0x06, 0x6100);
   snd_hda_set_pin_ctl_cache(codec, mic_pin, PIN_VREF50);
   break;
+ case 0x10ec0233:
  case 0x10ec0283:
   alc_write_coef_idx(codec, 0x45, 0xc429);
   snd_hda_set_pin_ctl_cache(codec, hp_pin, 0);
@@ -3636,6 +3638,7 @@ static void alc_headset_mode_default(struct hda_codec *codec)
   alc_write_coefex_idx(codec, 0x57, 0x03, 0x8ea6);
   alc_write_coef_idx(codec, 0x49, 0x0049);
   break;
+ case 0x10ec0233:
  case 0x10ec0283:
   alc_write_coef_idx(codec, 0x06, 0x2100);
   alc_write_coef_idx(codec, 0x32, 0x4ea3);
@@ -3677,6 +3680,7 @@ static void alc_headset_mode_ctia(struct hda_codec *codec)
   alc_write_coef_idx(codec, 0x1b, 0x0c2b);
   alc_write_coefex_idx(codec, 0x57, 0x03, 0x8ea6);
   break;
+ case 0x10ec0233:
  case 0x10ec0283:
   alc_write_coef_idx(codec, 0x45, 0xd429);
   alc_write_coef_idx(codec, 0x1b, 0x0c2b);
@@ -3715,6 +3719,7 @@ static void alc_headset_mode_omtp(struct hda_codec *codec)
   alc_write_coef_idx(codec, 0x1b, 0x0c2b);
   alc_write_coefex_idx(codec, 0x57, 0x03, 0x8ea6);
   break;
+ case 0x10ec0233:
  case 0x10ec0283:
   alc_write_coef_idx(codec, 0x45, 0xe429);
   alc_write_coef_idx(codec, 0x1b, 0x0c2b);
@@ -3757,6 +3762,7 @@ static void alc_determine_headset_type(struct hda_codec *codec)
   val = alc_read_coef_idx(codec, 0x46);
   is_ctia = (val & 0x0070) == 0x0070;
   break;
+ case 0x10ec0233:
  case 0x10ec0283:
   alc_write_coef_idx(codec, 0x45, 0xd029);
   msleep(300);
-- 
1.7.1